starfish/pipeline/features/spots/detector/detect.py

```diff
@@ -0,0 +1,222 @@
+from functools import partial
+from itertools import product
+from typing import Callable, Dict, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from starfish.constants import Indices, Features
+from starfish.image import ImageStack
+from starfish.intensity_table import IntensityTable
+from starfish.munge import dataframe_to_multiindex
+from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.typing import Number
+
+
+def measure_spot_intensity(
+        image: Union[np.ndarray, xr.DataArray],
+        spots: pd.DataFrame,
+        measurement_function: Callable[[Sequence], Number]) \
+        -> pd.Series:
+    """measure the intensity of each spot in spots in the corresponding image
+
+    Parameters
+    ----------
+    image : Union[np.ndarray, xr.DataArray],
+        3-d volume in which to measure intensities
+    spots : pd.DataFrame
+        SpotAttributes table containing coordinates and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
```

Comment:
no back slash.

also, does it take a sequence?  i thought it takes a numpy array.

Reply:
Since we're designing this around numpy ufuncs, they are actually more general -- you can feed them lists, pandas series, tuples, or pandas dataframes, and they'll all generate the correct output. We could choose to limit flexibility here as a design decision.